### PR TITLE
Implement index:info() introspection

### PR DESCRIPTION
**Pull request by rtsisyk**
Extract and refactor vy_info implementation from Vinyl in order to implement index:info() introspection.

```
tarantool> box.space.vinyl.index.primary:info()
---
- page_size: 8192
  size: 0
  run_avg: 0
  range_count: 1
  run_histogram: '[0]:1'
  page_count: 0
  run_count: 0
  memory_used: 3400
  count: 100
  range_size: 1073741824
...
```

Short history of previous reviews:

- [x] Comments to info.h headers by @kostja 
- [x] Replace `struct info_node` with virtual methods by @alyapunov 

See #1662

### CODE REVIEW

File: src/box/lua/info.c [hunk 1]
```diff
@@ -208,6 +208,11 @@ luaT_info_handler(struct info_node *node, void *ctx)
 	struct lua_State *L = ctx;
 
 	switch (node->type) {
+	case INFO_BEGIN:
+		lua_newtable(L);
+		break;
+	case INFO_END:
+		break;
```
Comment (locker) [inline src/box/lua/info.c]: Nit: this hunk rather belongs to the previous patch.

File: test/vinyl/ddl.result [hunk 2]
```diff
@@ -115,6 +115,8 @@ box.snapshot()
 ...
 while box.info.vinyl().db[space.id..'/0'].count ~= 0 do fiber.sleep(0.01) end
 ---
+- error: '[string "while box.info.vinyl().db[space.id..''/0''].cou..."]:1: attempt
+    to index field ''db'' (a nil value)'
```
Comment (locker) [inline test/vinyl/ddl.result]: You forgot to fix this test.

Comment (rtsisyk) [inline test/vinyl/ddl.result]: good catch, thanks.

File: src/box/vinyl.c [hunk 3]
```diff
@@ -5211,195 +5211,139 @@ static void vy_conf_delete(struct vy_conf *c)
...
-	}
-	vy_info_table_end(h);
+	info_table_begin(h, "metric");
+	info_append_u64(h, "lsn", env->xm->lsn);
+	info_table_end(h);
 }
 
-void
-vy_info_gather(struct vy_env *env, struct vy_info_handler *h)
+int
```
Comment (alyapunov) [inline src/box/vinyl.c]: I guess it must be void

File: src/box/vinyl.c [hunk 4]
```diff
@@ -5211,195 +5211,139 @@ static void vy_conf_delete(struct vy_conf *c)
...
+	info_begin(h);
 	vy_info_append_global(env, h);
 	vy_info_append_memory(env, h);
 	vy_info_append_metric(env, h);
 	vy_info_append_performance(env, h);
+	info_end(h);
+	return 0;
+}
+
+int
```
Comment (alyapunov) [inline src/box/vinyl.c]: I guess it must be void too

File: src/box/vinyl.c (hunk 3 above)
Comment (rtsisyk) [inline src/box/vinyl.c]: Fixed

File: src/box/vinyl.c (hunk 4 above)
Comment (rtsisyk) [inline src/box/vinyl.c]: Fixed

**Comment by rtsisyk** [comment]
>Wednesday, April 12, 2017 6:27 PM +03:00 from Aleksandr Lyapunov <notifications@github.com>:
>
>@alyapunov requested changes on this pull request.
>Almost perfect. I like the virtual table.
>I found one minor issue, see below.

Fixed.

-- 
WBR,
Roman Tsisyk <roman@tarantool.org>
http://tarantool.org/ - an efficient in-memory data store and a Lua application server

**Pull request merged**